prov/usnic/src/usdf.h

```diff
@@ -326,16 +326,31 @@ struct usdf_mr {
 	struct usd_mr *mr_mr;
 };
 
+struct usdf_soft_comp {
+	int status;
+	/* Add new fields as required */
+};
+
 struct usdf_cq_hard {
 	struct usdf_cq *cqh_cq;
 	struct usd_cq *cqh_ucq;
 	atomic_t cqh_refcnt;
 	void (*cqh_progress)(struct usdf_cq_hard *hcq);
-	void (*cqh_post)(struct usdf_cq_hard *hcq, void *context, size_t len);
+	void (*cqh_post)(struct usdf_cq_hard *hcq, void *context, size_t len,
+			struct usdf_soft_comp soft_comp);
 	TAILQ_ENTRY(usdf_cq_hard) cqh_link;
 	TAILQ_ENTRY(usdf_cq_hard) cqh_dom_link;
 };
 
+struct usdf_cq_soft_entry {
+	void		*op_context;
+	uint64_t	flags;
+	size_t		len;
+	void		*buf;
+	uint64_t	data;
+	int		status;
```

Comment:
All these fields should follow the convention of using a 2-3 letter prefix to facilitate grepping and tags/cscope usage.  I suggest `se_` or `cse_`.